library/ssl_tls13_generic.c

```diff
@@ -614,10 +618,54 @@ static int ssl_tls13_parse_certificate( mbedtls_ssl_context *ssl,
 static int ssl_tls13_validate_certificate( mbedtls_ssl_context *ssl )
 {
     int ret = 0;
+    int authmode = ssl->conf->authmode;
     mbedtls_x509_crt *ca_chain;
     mbedtls_x509_crl *ca_crl;
     uint32_t verify_result = 0;
 
+    /* If SNI was used, overwrite authentication mode
```

Comment:
Regarding `authmode` in TLS 1.3 on client side according to the spec we should support only MBEDTLS_SSL_VERIFY_REQUIRED. Thus we should enforce this and make it clear here. After initializing `authmode` to `MBEDTLS_SSL_VERIFY_REQUIRED` above, what about the following:
```C
#if defined(MBEDTLS_SSL_SRV_C)
    if( ssl->conf->endpoint == MBEDTLS_SSL_IS_SERVER )
    {
#if defined(MBEDTLS_SSL_SERVER_NAME_INDICATION)                                 
        if( ssl->handshake->sni_authmode != MBEDTLS_SSL_VERIFY_UNSET )              
            authmode = ssl->handshake->sni_authmode;
        else                                
#endif
            authmode = ssl->conf->authmode;
    }
#endif
```